File: search/search_javascript/sequence_search.js
/**
 * Created by Epulari T on 2018/8/13.
 */


/**
 * 顺序查找-不考虑重复
 * 依次扫描每个元素并与关键字比较
 *
 * @para = arr 数组
 * @para = key 关键字
 */
function SequenceSearch(arr, key) {
	var len = arr.length;
	//索引一定不会为负
	var k = 0;
	for(var i = 0; i < len; i++) {
		if(arr[i] == key) {
			console.log("比较第" + ++k + "次");
			return i;
		}
		++k;
	}
	console.log("比较第" + k + "次");
	return -1;
}

var arr = [7, 22, 23, 24, 26, 27, 28, 32, 33, 38, 53, 54, 78];
console.log("查找成功： " + SequenceSearch(arr, 53));
console.log("查找失败： " + SequenceSearch(arr, 52));
/**
 * result
	比较第11次
	查找成功： 10
	比较第13次
	查找失败： -1
 */

File: Csharp-SortSearch/Csharp-SortSearch/Search/SequenceSearch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Csharp_SortSearch.Search
{
    /*
     * 功能
     * 顺序查找
     * 依次扫描每个元素并与关键字比较
     * Epulari T 
     * 2018.8.13
     */
    public class SequenceSearch
    {
        /// <summary>
        /// 顺序查找-不考虑重复
        /// </summary>
        /// <param name="arr">数组</param>
        /// <param name="key">关键字</param>
        /// <returns>key对应的原数组索引</returns>
        public int MySequenceSearch(int[] arr, int key)
        {
            int len = arr.Length;
            //索引一定不会为负
            int k = 0;
            for (int i = 0; i < len; i++)
            {
                if(arr[i] == key)
                {
                    Console.WriteLine("比较第" + ++k + "次");
                    return i;
                }
                ++k;
            }
            Console.WriteLine("比较第" + k + "次");
            return -1;
        }
    }
}
